contracts/lifecycle/DestructibleDelayed.sol

```diff
@@ -0,0 +1,74 @@
+pragma solidity ^ 0.4.18;
+
+import "../ownership/Ownable.sol";
+
+/**
+ * @title DestructibleDelayed
+ * @dev Contract that can be selfdestructed after a delay period.
+ */
+contract DestructibleDelayed is Ownable {
+
+   // Whether a request for the self-destruction of the contract was requested.
+   bool public destructionRequested = false;
+
+   // When self-destruction is possible
+   uint256 public destructionTime = 2**256-1;
+
+   // Minimum delay after which destroy() can be called once requested
+   uint256 public SELFDESTRUCTION_DELAY; 
+
+   // Events
+   event SelfDestructionRequest(uint256 _destructionTime);
+   event SelfDestructionRequestCancelled();
+
+  /*
+     @dev Constructor that sets SELFDESTRUCTION_DELAY
+     @param _destructionDelay Imposed delay between destructionRequest and selfdestruct.
+  */
+  function DestructibleDelayed(uint256 _destructionDelay) public {
+    SELFDESTRUCTION_DELAY =_destructionDelay;
+  }
+
+   /*
+      @dev Requesting to destroy the contract 
+   */
+   function destroyRequest() public onlyOwner returns (bool)
+   {
+      require(!destructionRequested);
+
+      destructionRequested = true;
+      destructionTime = now + SELFDESTRUCTION_DELAY; 
+
+      SelfDestructionRequest(destructionTime);
+      return true;
```

Comment:
what's the benefit for returning true here as opposed to just throwing / not-throwing?

Reply:
Hum, well, the function will ever only return `true` if everything passes, otherwise will throw. I honestly added out of common habit (see ERC20 functions), which apparently is added to make it easy on the UI end, since calling it with web3 will return a `bool` instead of the transaction.

Reply:
Yeah, I'm still not super clear on which pattern is preferred when. As far as I know, the "return bool" pattern is used when 1) the function is designed to be called by a contract and 2) the function call can fail, but the calling contract wants the ability to handle that case without throwing itself.

Reply:
I agree with @Shrugs on the criteria.

I don't understand the web3.js comment about how it returns a boolean. As far as I can tell from the documentation it will also return the transaction hash. https://github.com/ethereum/wiki/wiki/JavaScript-API#returns-50

Reply:
transactions themselves will return the tx receipt, yeah; afaik transactions don't have return values at all